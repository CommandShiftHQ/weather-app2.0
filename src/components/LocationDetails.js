import React from "react";
import PropTypes from "prop-types";

const LocationDetails = ({ city, country }) => {
  return (
    <>
      <h1 className="location">
        {city}, {country}
      </h1>
    </>
  );
};

LocationDetails.propTypes = {
  city: PropTypes.string,
  country: PropTypes.string,
};

export default LocationDetails;
