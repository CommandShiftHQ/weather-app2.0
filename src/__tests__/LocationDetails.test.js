import React from "react";
import { render, cleanup } from "@testing-library/react";
import LocationDetails from "../components/LocationDetails";

afterEach(cleanup);

describe("LocationDetails", () => {
  const { asFragment } = render(
    <LocationDetails city="Manchester" country="UK" />
  );

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct city and location props", () => {
    const { getByText } = render(
      <LocationDetails city="Manchester" country="UK" />
    );

    expect(getByText("Manchester, UK")).toBeTruthy();
  });
});
