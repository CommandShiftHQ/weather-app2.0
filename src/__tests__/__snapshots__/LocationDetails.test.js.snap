// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="location"
  >
    Manchester, UK
  </h1>
</DocumentFragment>
`;
